### PR TITLE
Vet clinic database: query and update animals table

### PR DESCRIPTION
**Pull request by Log-benjamin**
- [ ] Use database transactions.
- [ ] Modify and delete data in SQL.
- [ ] Prepare complex queries that answer analytical questions.

1. Inside a transaction update the animals' table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.

2. Update the animals table by setting the species column to digimon for all animals that have a name ending in mon. Update the animals table by setting the species column to pokemon for all animals that don't have species already set.

3. Now, take a deep breath and... Inside a transaction delete all records in the animals table, then roll back the transaction. After the rollback verify if all records in the animals table still exists. After that, you can start breathing as usual ;)

4. Inside a transaction: Delete all animals born after Jan 1st, 2022., Create a savepoint for the transaction., Update all animals' weight to be their weight multiplied by -1., Rollback to the savepoint, Update all animals' weights that are negative to be their weight multiplied by -1., Commit transaction
Screenshot (41)
Screenshot (42)
Screenshot (43)
Screenshot (45)

5. How many animals are there?
Screenshot (47)

6. How many animals have never tried to escape?
Screenshot (48)

7. What is the average weight of animals?
Screenshot (49)

8. Who escapes the most, neutered or not neutered animals?
Screenshot (50)

9. What is the minimum and maximum weight of each type of animal?
Screenshot (51)

10. What is the average number of escape attempts per animal type of those born between 1990 and 2000?

Screenshot (52)

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
-VALUES ('Agumon', '2020-02-03', 0, true, 10.23),
-	   ('Gabumon', '2018-11-15', 2, true, 8),
-	   ('Pikachu', '2021-01-07', 1, false, 15.04),
-	   ('Devimon', '2017-05-12', 5, true, 11)
\ No newline at end of file
+INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES (1, 'Agumon', '2020-02-03', 0, true, 10.23),
+	   (2, 'Gabumon', '2018-11-15', 2, true, 8),
+	   (3, 'Pikachu', '2021-01-07', 1, false, 15.04),
+	   (4, 'Devimon', '2017-05-12', 5, true, 11)
```
Comment (Qoosim) [inline data.sql:5]: - [ ] Great job so far. However, there's small bug here. If your statement is not terninated with semi-column, I wouldn't be able to populate your table withh these data. I have actually added it to test your db and everything seems ok. But you need to correct that from your side too. I reached out to you on slack, but it appears you're not awere you have messages. Kindly add that and you will be good to move on to the next milestone.

File: data.sql [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+
+
+INSERT INTO animals(id, name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES (5, 'Charmander', '2020-02-08', 0, false, -11),
+	   (6, 'Plantmon', '2021-11-15', 2, true, -5.7),
+	   (7, 'Squirtle', '1993-04-02', 3, false, -12.13),
+	   (8, 'Angemon', '2005-06-12', 1, true, -45),
+	   (9, 'Boarmon', '2005-06-07', 7, true, 20.4),
+	   (10, 'Blossom', '1998-10-13', 3, true, 17),
+	   (11, 'Ditto', '2022-05-14', 4, true, 22)
```
Comment (Qoosim) [inline data.sql:15]: - [ ] Same thing here.

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @Log-benjamin, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  Table updated and queried correctly. :heavy_check_mark:  
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by fobadara — APPROVED** [review]
Hi @Log-benjamin, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

APPRVED
## Highlights
✅ Descriptive README
✅ Descriptive PR summary
✅ Clean code

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**